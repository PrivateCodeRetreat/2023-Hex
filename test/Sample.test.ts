import {describe, test} from "@jest/globals";
import {verify} from "approvals/lib/Providers/Jest/JestApprovals";

class GameOfLifeBoard {
    private cells: [number, number][];

    constructor(...tuples: [number, number][]) {
        this.cells = tuples;
    }

    printHexBoard(width: number, height: number, printCoordinates: boolean): string {
        // The pieces that make up each hex cell
        let grid = this.printHeader(width);
        for (let row = 0; row < Math.ceil(height / 2); row++) {
            let oddY = (row + 1) * 2 - 1;
            grid = this.printOddRows(width, grid, oddY, printCoordinates);
            grid = this.printEvenRows(width, grid, oddY + 1, printCoordinates);
        }

        return grid;
    }

    private printHeader(width: number) {
        let grid = '';
        const header = " __   ";
        for (let col = 0; col < width / 2; col++) {
            grid += header;
        }

        return grid.substring(0, grid.length - 2) + "\n";
    }

    private printEvenRows(width: number, grid: string, y: number, printCoordinates: boolean) {
        const empty = "\\__/  ";
        const alive = "\\__/##";

        for (let x = 2; x <= width; x += 2) {
            if (this.isCellAlive([x, y])) {
                console.log(`alive at [${x}, ${y}]`);
                grid += alive
            } else {
                if (printCoordinates) {
                    grid += `\\__/${x % 10}${y % 10}`;
                } else {
                    grid += empty;
                }
            }
        }

        return grid.substring(0, grid.length - 2) + "\n";
    }


    private printOddRows(width: number, grid: string, y: number, printCoordinates: boolean) {
        const empty = "/  \\__";
        const alive = "/##\\__";

        for (let x = 1; x <= width; x += 2) {
            if (this.isCellAlive([x, y])) {
                grid += alive
            } else {
                if (printCoordinates) {
                    grid += `/${x}${y}\\__`;
                } else {
                    grid += empty;
                }
            }
        }

        return grid.substring(0, grid.length - 2) + "\n";
    }

    toString() {
        return this.printHexBoard(10, 10, false);
    }

    private isCellAlive(cell: [number, number]) {
        return this.cells.toString().includes(cell.toString());
    }
}

describe("ApprovalTests", () => {

    test("Print game of life board", () => {

        verify(new GameOfLifeBoard());
    });

    test("Testing coordinates", () => {

        verify(new GameOfLifeBoard([1, 1], [2, 2], [3, 5], [4, 6]));
        // verify(new GameOfLifeBoard([2, 2], [3, 5], [4, 6]));
    });

    test("Test board with numbers", () => {

        verify(new GameOfLifeBoard().printHexBoard(10, 10, true));
    });
});
