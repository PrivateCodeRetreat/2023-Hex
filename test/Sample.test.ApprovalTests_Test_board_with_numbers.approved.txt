 __    __    __    __    __ 
/11\__/31\__/51\__/71\__/91\
\__/22\__/42\__/62\__/82\__/
/13\__/33\__/53\__/73\__/93\
\__/24\__/44\__/64\__/84\__/
/15\__/35\__/55\__/75\__/95\
\__/26\__/46\__/66\__/86\__/
/17\__/37\__/57\__/77\__/97\
\__/28\__/48\__/68\__/88\__/
/19\__/39\__/59\__/79\__/99\
\__/20\__/40\__/60\__/80\__/
